torch/_inductor/codegen/cpp.py

```diff
@@ -831,6 +832,28 @@ def index_expr(expr, dtype):
         assert opt_ctx.is_most_inner_loop_irrevelant
         return f"at::vec::Vectorized<int>(static_cast<int>({cexpr(V.kernel.rename_indexing(expr))}))"
 
+    def check_bounds(self, expr, size):
+        size = V.kernel.rename_indexing(size)
+
+        lower = V.kernel.cse.newvar()
+        upper = V.kernel.cse.newvar()
+
+        code = IndentedBuffer()
+
+        index_expr = V.kernel.rename_indexing(expr)
+        code.writeline(f'auto {lower} = {cexpr(index_expr)};')
+        code.writeline(f'auto {upper} = {cexpr(index_expr)};')
+
+        x = V.kernel.cse.newvar()
+        code.writeline(f'auto {x} = (0 <= {lower}) & ({upper} < {size});')
```

Comment:
Why can't we have an indirect assert line?
Also, we would like to share the logic here and in triton where we write one of the bounds (or none) if we can prove that the expression is within bounds.

Reply:
I've changed the code to reflect that, Mario. Can you please review once you have a chance?